external-builds/pytorch/pytorch_smoke_test_launcher.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+import pytest
+import sys
+import os
+import subprocess
+from pathlib import Path
+
+
+def get_executable_pytorch_gpu_index_list() -> list[str]:
+    """Get index number list for gpus that are available on computer
+       and are also supported by the currently tested pytorch build.
+       Returned gpu index numbers can be used for controlling the GPU
+       visibility via HIP_VISIBLE_DEVICES environment variable.
```

Comment:
As is done in https://github.com/ROCm/TheRock/pull/2468, code like this should be shared between scripts